### PR TITLE
expr eval - return error if parsing failed

### PR DESCRIPTION
**Pull request by justcoon**
related to: #688

### CODE REVIEW

File: golem-rib/src/text/mod.rs [hunk 1]
```diff
@@ -425,6 +425,25 @@ mod record_tests {
         let output_expr = from_string(expr_str.as_str()).unwrap();
         assert_eq!((expr_str, input_expr), (expected_record_str, output_expr));
     }
+
+    #[test]
+    fn test_round_trip_record_literal_invalid() {
+        let expr_str = r#"
+                 ${  {body: golem:component/api.{get-character}(), headers: { x-test: 'foobar' } } }
```
Comment (afsalthaj) [inline golem-rib/src/text/mod.rs:432]: The existence of this test is not a big problem, however I wouldn't personally include this test if was doing it. So its very opinionated statement.
 
All other tests in golem-rib is structured (as much as I can) and has a behavior. Example: testing a  `record`parser will test `record of primitive` or `record of sequence`, or `record of record` etc.  Infact, this is why I wrote things from scratch instead of copying old tests for this parser because old tests are all "randomish".

And on the other hand this particular, this test is very specific - "an invalid case of a record having a value which is having single quotes",  such tests can be theoretically infinite. This opens door to having "random" tests to exist in golem-rib.

File: golem-worker-service-base/src/api_definition/http/http_api_definition.rs [hunk 2]
```diff
@@ -379,27 +379,27 @@ mod tests {
 
     #[test]
     fn expression_with_predicate0() {
-        test_string_expr_parse_and_encode("1>2");
+        test_string_expr_parse_and_encode("${ 1 < 2 }");
```
Comment (afsalthaj) [inline golem-worker-service-base/src/api_definition/http/http_api_definition.rs:382]: why was this changed?

File: golem-worker-service-base/src/api_definition/http/http_api_definition.rs [hunk 3]
```diff
@@ -379,27 +379,27 @@ mod tests {
...
     }
 
     #[test]
     fn expression_with_predicate1() {
-        test_string_expr_parse_and_encode("${request.path.user-id > request.path.id}");
+        test_string_expr_parse_and_encode("${ request.path.user-id > request.path.id }");
```
Comment (afsalthaj) [inline golem-worker-service-base/src/api_definition/http/http_api_definition.rs:387]: why changing thse tests?

File: golem-worker-service-base/src/api_definition/http/http_api_definition.rs [hunk 4]
```diff
@@ -379,27 +379,27 @@ mod tests {
...
     #[test]
     fn expression_with_predicate3() {
-        test_string_expr_parse_and_encode("${request.path.user-id}=2");
+        test_string_expr_parse_and_encode("${ request.path.user-id == 2 }");
     }
 
     #[test]
     fn expression_with_predicate4() {
-        test_string_expr_parse_and_encode("${request.path.user-id}<2");
+        test_string_expr_parse_and_encode("${ request.path.user-id < 2 }");
```
Comment (afsalthaj) [inline golem-worker-service-base/src/api_definition/http/http_api_definition.rs:402]: Should there be a space now?  or the actual fix is the curly bracce had to go out?

File: golem-worker-executor-base/src/invocation.rs [hunk 5]
```diff
@@ -265,7 +265,7 @@ async fn get_or_create_indexed_resource<'a, Ctx: WorkerCtx>(
         )),
     )?;
 
-    let constructor_param_types = resource_constructor.params(&store).iter().map(type_to_analysed_type).collect::<Result<Vec<_>, _>>()
+    let constructor_param_types = resource_constructor.params(store as &StoreContextMut<'a, Ctx>).iter().map(type_to_analysed_type).collect::<Result<Vec<_>, _>>()
```
Comment (afsalthaj) [inline golem-worker-executor-base/src/invocation.rs:268]: Clippy driven?

**Review by afsalthaj — APPROVED** [review]

File: golem-worker-service-base/src/api_definition/http/http_api_definition.rs (hunk 2 above)
Comment (justcoon) [reply golem-worker-service-base/src/api_definition/http/http_api_definition.rs:382]: because it was failing on
```
called `Result::unwrap()` on an `Err` value: "Failed to parse: \"1 < 2\""
```

File: golem-worker-service-base/src/api_definition/http/http_api_definition.rs (hunk 4 above)
Comment (justcoon) [reply golem-worker-service-base/src/api_definition/http/http_api_definition.rs]: curly brace has to go out

otherwise
```
called `Result::unwrap()` on an `Err` value: "Failed to parse: \"${request.path.user-id}<2\""
```
